### PR TITLE
Add UMI-tool extract

### PR DESCRIPTION
**Pull request by blankclemens**
> Extract UMI barcode from a read and add it to the read name, leaving any sample barcode in place. Can deal with paired end reads and UMIs split across the paired ends.

This is only the extract tool from the UMI-tools suite, other parts may follow.

### CODE REVIEW

**Comment by davebx** [comment]
Ooh, this is exactly what I need to finish generating proper test data for `kallisto pseudo --umi`

Thanks for the work, @blankclemens

File: tools/umi_tools/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+categories:
+- Sequence Analysis
+- Transcriptomics
+description: UMI-tools extract - Extract UMIs from fastq
+homepage_url: https://github.com/CGATOxford/UMI-tools
+long_description: |
+  Extract UMI barcode from a read and add it to the read name, leaving
+  any sample barcode in place. Can deal with paired end reads and UMIs
+  split across the paired ends
+name: umi-tools_extract
```
Comment (bgruening) [inline tools/umi_tools/.shed.yml]: If the other tools might follow, maybe you want to make a suite definition here already?

File: tools/umi_tools/umi-tools_extract.xml [hunk 2]
```diff
@@ -0,0 +1,306 @@
+<tool id="umi-tools_extract" name="UMI-tools extract" version="0.4.4.0">
+    <description>Extract UMI from fastq(.gz) files</description>
```
Comment (bgruening) [inline tools/umi_tools/umi-tools_extract.xml]: I would remove the (,gz) here, a user should not care about this detail.

Comment (blankclemens) [inline tools/umi_tools/umi-tools_extract.xml]: Indeed, thanks!

File: tools/umi_tools/.shed.yml (hunk 1 above)
Comment (blankclemens) [inline tools/umi_tools/.shed.yml]: Would there be any advantage at the moment compared to making it a suite with the second tool?

File: tools/umi_tools/umi-tools_extract.xml [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                #end if
+            #end if
+            #if not $prime3:
+                --3prime
+            #end if
+            #if $quality.quality_selector =='true':
+                --quality-filter-threshold '$quality.quality_filter_threshold'
+                --quality-encoding '$quality.quality_encoding'
+            #end if
+            --log='$out_log'
```
Comment (bgruening) [inline tools/umi_tools/umi-tools_extract.xml]: Can this be disabled if not used?

File: tools/umi_tools/umi-tools_extract.xml [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        </param>
+        <param name="prime3" argument="--3prime" type="boolean" label="Is the barcode at the 5' end?"
+            truevalue="1" falsevalue="0" checked="true"
+            help="By default the barcode is assumed to be on the 5' end of the read, but
+                use this option to sepecify that it is on the 3' end instead." />
+        <param name="print_log" type="boolean" label="Output log?"
+            truevalue="1" falsevalue="0" checked="true"
+            help="Choose if you want to generate a text file containing logging information." />
+        <conditional name="quality">
+            <param name="quality_selector" type="select" label="Enable quality filter?" >
```
Comment (bgruening) [inline tools/umi_tools/umi-tools_extract.xml]: This looks like a ideal candidate for a `section` :)

File: tools/umi_tools/.shed.yml (hunk 1 above)
Comment (bgruening) [inline tools/umi_tools/.shed.yml]: the creation of repositories are different ... the names are generated out of ID ... so tool IDs are in sync with the repo names.

Comment (blankclemens) [inline tools/umi_tools/.shed.yml]: Done, thanks!

File: tools/umi_tools/umi-tools_extract.xml (hunk 4 above)
Comment (blankclemens) [inline tools/umi_tools/umi-tools_extract.xml]: While I am a big fan of sections, I don't see any advantage here. I like them for big, often used settings that would ruin the tool interface if always displayed. Here we have two options which do not identify as the core utility of this tool.

**Pull request merged**